From 437a2e7493666da4643f16933082372df1774b57 Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Date: Wed, 10 Jan 2018 11:52:54 -0600
Subject: [PATCH 3181/5242] MLK-17363-2 imx8: pm-domain: clocks list should be
 initialized only once.

commit  9fbc6cc9e6e8222d3ccfeb5760ed132713413ea9 from
https://source.codeaurora.org/external/imx/linux-imx.git

The clocks list associated with a PD is the same across all devices
attached to the same PD. Re-initializing it each time a new device is
attached results in missing some clocks.

[ Aisheng: "Improve commit message" ]

Reviewed-by: Anson Huang <anson.huang@nxp.com>
Signed-off-by: Dong Aisheng <aisheng.dong@nxp.com>
Signed-off-by: Ranjani Vaidyanathan <Ranjani.Vaidyanathan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/pm-domains.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/soc/imx/pm-domains.c b/drivers/soc/imx/pm-domains.c
index 4b044dc..1431e51 100644
--- a/drivers/soc/imx/pm-domains.c
+++ b/drivers/soc/imx/pm-domains.c
@@ -205,7 +205,6 @@ static int imx8_attach_dev(struct generic_pm_domain *genpd, struct device *dev)
 	int rc, index, num_clks;
 
 	pd = container_of(genpd, struct imx8_pm_domain, pd);
-	INIT_LIST_HEAD(&pd->clks);
 
 	num_clks = of_count_phandle_with_args(node, "assigned-clocks",
 						"#clock-cells");
-- 
1.7.9.5

